x-pack/plugins/apm/server/lib/errors/get_error_rate.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ProcessorEvent } from '../../../common/processor_event';
+import {
+  ERROR_GROUP_ID,
+  SERVICE_NAME,
+  PROCESSOR_EVENT,
+} from '../../../common/elasticsearch_fieldnames';
+import { getBucketSize } from '../helpers/get_bucket_size';
+import { rangeFilter } from '../helpers/range_filter';
+import {
+  Setup,
+  SetupTimeRange,
+  SetupUIFilters,
+} from '../helpers/setup_request';
+
+export async function getErrorRate({
+  serviceName,
+  groupId,
+  setup,
+}: {
+  serviceName: string;
+  groupId?: string;
+  setup: Setup & SetupTimeRange & SetupUIFilters;
+}) {
+  const { start, end, uiFiltersES, client, indices } = setup;
+  const { intervalString } = getBucketSize(start, end, 'auto');
+  const groupIdTerm = groupId ? [{ term: { [ERROR_GROUP_ID]: groupId } }] : [];
+
+  const filter = [
+    { term: { [SERVICE_NAME]: serviceName } },
+    { range: rangeFilter(start, end) },
+    ...uiFiltersES,
+  ];
+
+  const aggs = {
+    response_times: {
+      date_histogram: {
+        field: '@timestamp',
+        fixed_interval: intervalString,
+        min_doc_count: 0,
+        extended_bounds: { min: start, max: end },
+      },
+    },
+  };
+
+  const getTransactionBucketAggregation = async () => {
+    const resp = await client.search({
+      index: indices['apm_oss.transactionIndices'],
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...filter,
+              { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
+            ],
+          },
+        },
+        aggs,
+      },
```

Comment:
why would I use `track_total_hits`?

Reply:
you might know this, but: if you don't use it ES will only count 10k docs. you can set it to `true` which means it will count all docs or use a number to set a limit of docs to count. In this case it doesn't matter because you're only checking if there is more than 0 hits, and in terms of performance you're doing an aggregation so ES will process all documents that match the search anyway.

Reply:
Does it also impacts the aggregations? because indeed I'm only using the `hits.total.value` to check if the transaction is greater than 0. But in the aggregation, I need the total of transactions and errors in a given interval, which can be more than 10k.

Reply:
It doesn't. The default behaviour (ES will stop after finding 10000 docs if there are no sort options and aggregations) is a performacne optimization. However, if there _are_ sort options and/or aggregations, ES will process the entire data set anyway. AFAIK, only `terminate_after` and `timeout` will impact what documents are processed by the aggregations.

Reply:
cool, thanks for the explanation. In that case, I don't need the `track_total_hits`.